### ISSUE TITLE
Bug Report: "Total notes" on the results screen does not reflect the actual number of notes in a song.

### ISSUE THREAD

**Issue by NewCs386148** [FunkinCrew/Funkin#3016]
### Issue Checklist

- [X] I have properly named the issue
- [X] I have checked the issues/discussions pages to see if the issue has been previously reported

### What platform are you using?

Newgrounds (Web)

### If you are playing on a browser, which one are you using?

Microsoft Edge

### Version

0.4.1

### Context (Provide images, videos, etc.)

I played 2hot and ended up getting 89% despite my "sicks" and "goods" totaling to about 91% of the total notes. I added up the numbers of each category, and it turns out that somehow there are 17 more notes there than there are in the total notes section of the screen.

Score:
![2hotScore](https://github.com/user-attachments/assets/60638dc4-f51e-46d5-940e-f7eca0488598)

Math:
![2hotMath](https://github.com/user-attachments/assets/1250c46e-1eff-47e5-9f30-5aae69d0d0e3)

After playing the song again, the "total notes" was an entirely different number! However, adding up the numbers of each individual accuracy category still adds up to 585, and the % cleared still uses 585 as the number of notes in the song to calculate its percentage.
![2hotScore2](https://github.com/user-attachments/assets/89d5892b-3529-49bb-ba2d-bbf4a373faa8)
![2hotMath2](https://github.com/user-attachments/assets/17947f2b-b157-47ec-9be7-cbfb3b120b31)

### Steps to reproduce (or crash logs, errors, etc.)

Play any song in the game and don't get a full combo. In this example here, I played the tutorial on normal and missed the whole first turn.
![TutorialLMAO](https://github.com/user-attachments/assets/ba40f845-60b0-4674-bee8-6b8f57ea2d03)

**Comment by Hundrec** [comment]
This is a really strange bug that seems to only affect the displayed value for "Total Notes". 

From your report, I'm getting that none of the calculations related to the total number of notes were affected, right?

**Comment by Keoiki** [comment]
Based on the 3rd picture, you'd think Total Notes means the total notes you **HIT** in the song, not total notes in the song in general.

But putting together the numbers, excluding misses, gives 566 in the first picture (2 lower than what Total Notes says) and 561 in the second picture (1 lower than what Total Notes says)

What gives?

**Comment by NewCs386148** [comment]
>From your report, I'm getting that none of the calculations related to the total number of notes were affected, right?

Yes. Despite the displayed number being wrong, it still calculates percentages with the real number of total notes (which seems to be the number of all five accuracy categories added up).

**Comment by Hundrec** [comment]
Maybe there's a rare chance for the game to skip counting a hit note if there's a lagspike?

**Comment by NewCs386148** [comment]
Possibly. I find it strange that the game seems to calculate total notes while you're playing the song instead of deriving that value from the chart itself.

**Comment by Keoiki** [comment]
OK This isn't even a bug, it's like that on purpose as that number shows the `totalNotesHit`:
https://github.com/FunkinCrew/Funkin/blob/9b483de1da9c1b72b02ec30f4d13923470cf32b9/source/funkin/play/ResultState.hx#L368
https://github.com/FunkinCrew/Funkin/blob/9b483de1da9c1b72b02ec30f4d13923470cf32b9/source/funkin/play/ResultState.hx#L371

But I still don't understand why either: the total number is wrong or the sick/good/bad/shit numbers are wrong (based on what I said in my previous comment)

**Comment by Hundrec** [comment]
So that leaves the off-by-a-few issue as well as a need for clarification that it's "Total Notes **Hit**" rather than "Total Notes in the song"

**Comment by NewCs386148** [comment]
>a need for clarification that it's "Total Notes Hit" rather than "Total Notes in the song"

They could make it say something like "Total Notes: [Total notes hit]/[Total notes in song]" instead of "Total Notes: [Total notes hit]". They'd have to make it a scrolling textbox or something to keep it off the yellow background, but it would properly communicate both values.

**Comment by lemz1** [comment]
> But I still don't understand why either: the total number is wrong or the sick/good/bad/shit numbers are wrong (based on what I said in my previous comment)


https://github.com/FunkinCrew/Funkin/blob/9b483de1da9c1b72b02ec30f4d13923470cf32b9/source/funkin/play/PlayState.hx#L2469-L2471

I believe this could be the issue for notes not being accounted for when hit. Maybe for some reason `event.eventCanceled` is true which skips over the rest.

**Comment by lemz1** [comment]
https://github.com/FunkinCrew/Funkin/blob/9b483de1da9c1b72b02ec30f4d13923470cf32b9/source/funkin/play/PlayState.hx#L2468-L2480

The issue might be that `event.eventCanceled` is true, which skips over `applyScore`. applyScore handles the sick/good/bad/shit counter

**Comment by lemz1** [comment]
yup im pretty sure that is the case.
in `2hot.hxc` there is this part:
```haxe
case "weekend-1-firegun":
  if (gunCocked)
  {
	  trace('Firing gun!');
	  shootNextCan();
  }
  else
  {
	  trace('Cannot fire gun!');
	  // The player cannot hit this note.
	  event.cancelEvent();
  }
```

i imagine what happened is that they hit the note too early, which lead to the event getting canceled, which in turn meant that it didnt update the sick/good/bad/shit counter

**Comment by Lasercar** [comment]
Yeah, total notes = total notes _hit_.

**Comment by amyspark-ng** [comment]
isn't that weird?

**Comment by Hundrec** [comment]
Just to be sure, did we decide this was due to the Total Notes vs. Total Notes **Hit** quirk, and that nothing is being miscalculated here?

**Comment by VioletSnowLeopard** [comment]
> Just to be sure, did we decide this was due to the Total Notes vs. Total Notes **Hit** quirk, and that nothing is being miscalculated here?

I think so. I believe that Total Notes reflects how many you actually hit in the song. Regarding the off-by-a-few issue, I think that is fixed now. I tested playing several songs, and all notes hit (ignoring misses) added up to the total notes shown on the results.

**Comment by Hundrec** [comment]
Nice, let's hope this one's gone for good (unlike the song restarting bug)
